backend/models/loginRecord.js

```diff
@@ -0,0 +1,127 @@
+/**
+ *	Copyright (C) 2014 3D Repo Ltd
+ *
+ *	This program is free software: you can redistribute it and/or modify
+ *	it under the terms of the GNU Affero General Public License as
+ *	published by the Free Software Foundation, either version 3 of the
+ *	License, or (at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU Affero General Public License for more details.
+ *
+ *	You should have received a copy of the GNU Affero General Public License
+ *	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ "use strict";
+ const db = require("../handler/db");
+ 
+ //detects edge as browser but not device
+ var uaParserJs = require('ua-parser-js');
+ var device = require('device');
+ var ip2location = require('ip-to-location');
+
+ const LoginRecord = {};
+ 
+ LoginRecord.saveLoginRecord = async (req) => {
+        var loginRecord = {
+                _id : req.sessionID,
+                loginTime : new Date(),
+                ipAddr: req.ips[0] || req.ip                    
+        }
+
+        const referrer = req.header('Referer');
+        if(referrer != null){
+                loginRecord.referrer = referrer;
+        }
+
+        const location = await getLocationFromIPAddress(loginRecord.ipAddr);   
+        loginRecord.location = {
+                country: location.country_name,
+                city: location.city
+        };    
+       
+         const userAgentString = req.headers['user-agent'];
+         var uaInfo = {};
+
+         if(isUserAgentFromPlugin(userAgentString)){
+                uaInfo = getUserAgentInfoFromPlugin(userAgentString);
+         }
+         else{
+                uaInfo = getUserAgentInfoFromBrowser(userAgentString);
+         }
+
+         loginRecord.application = uaInfo.application;
+         loginRecord.engine = uaInfo.engine;
+         loginRecord.os = uaInfo.os;
+         loginRecord.device = uaInfo.device;          
+
+         const col = await db.getCollection("loginRecords", req.body.username);
```

Comment:
avoid getting your own collection object to do the insert, call [this ](https://github.com/3drepo/3drepo.io/blob/master/backend/handler/db.js#L184) in db.js

The idea is that if we abstract ourselves away from the db driver then it'd make our life easier should they change their API in future upgrades, also less of these double awaits everywhere

so you should be able to just do this
```suggestion
    await db.insert("loginRecords", req.body.username, loginRecord);
```